packages/video_player_videohole/lib/video_player_platform_interface.dart

```diff
@@ -271,7 +276,7 @@ class VideoEvent {
   /// Duration of the video.
   ///
   /// Only used if [eventType] is [VideoEventType.initialized].
-  final Duration? duration;
+  final DurationRange? duration;
```

Comment:
https://github.com/flutter/packages/blob/main/packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart#L229
Can't we just keep `Duration` type and use the value of `duration.second` returned by mediaplayer?
The interface that returns `DurationRange` for stream play seems to be `buffered`. (https://github.com/flutter/packages/blob/main/packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart#L244)
what do you think?

Reply:
For live stream, the duration is a range which includes start time and end time.
```C++
 player_get_adaptive_streaming_info(player_, (void *)&live_duration_buff, PLAYER_ADAPTIVE_INFO_LIVE_DURATION);
```
When call seekTo, the time must be in the range， the start time is only used to judge. If remove start time, we might have seekTo issue.
I have tested seekTo for live stream, if we get live stream duration is [300, 500] and then call seekTo(200),  the player just start with 300.

Reply:
My worry is the change in interface. (I know, we have a separate interface from video_player_platform_interface.)

In my opinion..
When a user needs a seekTo point in valid range of streaming content, they can check the valid seekTo by taking the value from `buffered` rather than the duration value.
https://github.com/flutter/packages/blob/36a7b99381f85e86914e82c75fc7d9038ed96cca/packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart#L241-L244
```dart
  /// Buffered parts of the video.
  ///
  /// Only used if [eventType] is [VideoEventType.bufferingUpdate].
  final List<DurationRange>? buffered;
```
Our `int buffered` is progress value of buffering. so we can change this API.
Of course, we don't necessarily have to follow the video_player_platform_interface, so this can be reconsidered later.